6/README.md

```diff
@@ -205,45 +228,110 @@ The endpoint must accept
 - POST requests to the `/async/operation-id` (e.g. https://service-endpoint/async/4d517500da0acb0d65a716f61330969334630363ce4a6a9d39691026ac7908ea) along with JSON formatted payload as described by the params section of the operation metadata.
 
 - The path argument operation-id must be the ID of the operation asset.
-- The keys in the (payload) map must be parameter names as specified in the operation metadata.
-- All required parameters must be included.
-
-- The values must be one of 
-  - A map (if the type is **asset**).
-    - The map must contain the DID (the Decentralized ID) against the key **did** 
-  - A valid JSON value (if the type is **json**). Here's the list of [valid JSON data types](https://json-schema.org/understanding-json-schema/reference/type.html)
+- The keys in the (payload) map may be parameter names as specified in the operation metadata.
+- All required parameters may be included.
+
+- The values can be any of following:
+   OpenAPI SchemaObject [OpenAPI Schema Object standards](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#schema-object).
+   This may include the following OpenAPI schema data types:
+        object
+        array
+        integer
+        number
+        string
+        boolean
+        
+- The parameters may be used to display swagger interface documentation.
 
 #### Design considerations
 
 This section is non-normative.
 
-The values are categorised into two types (`asset` and `json`) to support libraries such as Starfish in:
+* The `string` data type mave have the extended fields:
+    - x-resolve
+    - x-format
+
+
+* The **x-resolve** and **x-format** must have one of the following values:
+    - did-asset
+    - did
+
+- **did-asset** describes the type asset as a did/asset. For example
+
+
+```
+    # full asset id
+    
+    did:op:4d517500da0acb0d65a716f61330969334630363ce4a6a9d39691026ac7908ea/0xf8c12e287be8199a8c12eb9bbb63bc302c318b2290ef1e40af5a934e3315022d
+```
+
+- **did** describes a single did agent. For example
+
+```
+    # did resolve to an agent
+    
+    did:op:4d517500da0acb0d65a716f61330969334630363ce4a6a9d39691026ac7908ea
+    
+```
+
+- **x-resolve** Trys to resolve the asset id or did depending on the reference value.
+
+- **x-format** Validates the string value to make sure the string value is the correct format ( did or did-asset)
```

Comment:
This seems like too much complexity for the operation metadata standard. If the creator of an operation wishes to do custom validation, I suggest that they simply define a `json` parameter type and then do whatever validation they want on that.